core/src/main/scala/org/apache/spark/util/EventLoop.scala

```diff
@@ -100,7 +100,11 @@ private[spark] abstract class EventLoop[E](name: String) extends Logging {
    * Put the event into the event queue. The event thread will process it later.
    */
   def post(event: E): Unit = {
-    eventQueue.put(event)
+    if (eventThread.isAlive && !stopped.get) {
+      eventQueue.put(event)
+    } else {
+      onError(new IllegalStateException(s"$name has already been stopped."))
+    }
```

Comment:
here we only need to report thread stopped. There is no exception messages, but it is fine because thread will throw exception when exit.

Reply:
What about just ignoring the post? does it need to throw an error here?

Reply:
@srowen As these threads were not alive, reporting an error is to do some cleaning work gracefully. For example, when `JobGenerator` and `JobScheduler` is down, we will stop streaming context.

Reply:
Can this be 
```
if (!stopped.get) {
  if (eventThread.isAlive) {
    ...
  } else {
    ...
  }
}
```
OK I see that it won't fail if you post() when stopped. But it will throw an exception if it's not stopped but the event thread is dead.

Reply:
fixed